sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -1142,6 +1142,52 @@ object SimplifyCaseConversionExpressions extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * Removes date and time related functions that are unnecessary.
+ */
+object SimplifyDateTimeConversions extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan.transformWithPruning(
+    _.containsPattern(DATETIME), ruleId) {
+    case q: LogicalPlan => q.transformExpressionsUpWithPruning(
+      _.containsPattern(DATETIME), ruleId) {
```

Comment:
ditto. nit. Indentation?

Reply:
Fixed in https://github.com/apache/spark/commit/dcf49a839962c1aa6e6bc46ea0f61aed8f7f627a.